lib/WUI/http/httpd.c

```diff
@@ -2467,20 +2474,53 @@ void httpd_init(void) {
     #endif
     LWIP_DEBUGF(HTTPD_DEBUG, ("httpd_init\n"));
 
-    /* LWIP_ASSERT_CORE_LOCKED(); is checked by tcp_new() */
-
-    httpd_pcb = altcp_new_ip_type(&allocator, IPADDR_TYPE_ANY);
-    LWIP_ASSERT("httpd_init: tcp_new failed", httpd_pcb != NULL);
     httpd_reinit();
 }
 
+// The inner part of httpd_close. Assumes the mutex is already locked.
+static void httpd_close_locked() {
+    // According to docs, the close can fail in case there's not enough memory.
+    // That's likely because usual sockets still wait for some more data to
+    // arrive (?). Hopefully this is not the case for listening sockets.
+    //
+    // Using altcp_abort would be better in theory (as it just wipes it and
```

Comment:
LwESP does not seem to have abort counterpart. Unfortunately every ESP action allocates message structure -> can fail, also the ESP module may fail to fulfill the request -> there are multiple reasons this may fail.

In case of the ESP it may be possible to reset the device and reinitialize the LwESP. Anyway we may be thinking of deinit and reinit the library to save some RAM. Anyway, as of now I am fine with proposed solution using altcp_close.